### ISSUE TITLE
Optional default value

### ISSUE THREAD

**Issue by Odonno** [leny/react-use-storage#142]
I believe that the second parameter `defaultValue` should be optional (set as `undefined` by default so).

If I use it in my typescript app, I need to do that:

```ts
const [x, setX] = useLocalStorage<string | undefined>(KEY, undefined);
```

While what I want is to do that: 

```ts
const [x, setX] = useLocalStorage<string | undefined>(KEY);
```

I got time to read your ts definition, so I propose you this one:

```ts
export type StorageHook<V> = [V, (value: V) => void, () => void];

export declare const useLocalStorage: <V>(
    key: string,
    defaultValue?: V,
) => StorageHook<V>;

export declare const useSessionStorage: <V>(
    key: string,
    defaultValue?: V,
) => StorageHook<V>;
```

I also refactored the type definition a little.

**Comment by leny** [comment]
Nice!
Can you please make a Pull Request with your changes?

I will merge that tomorrow, I think.

**Comment by Odonno** [comment]
Done.

**Comment by leny** [comment]
Thanks. I have a lot to do today, will really try to merge that tomorrow, or friday.

**Comment by leny** [comment]
Published as `0.4.2`
